ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -81,27 +84,36 @@ def discover_queues(self, queue_manager, config):
         queues = []
         if config.auto_discover_queues:
             queues.extend(self._discover_queues(queue_manager, '*'))
-
-        if len(config.queue_patterns) > 0:
-            for regex in config.queue_patterns:
-                queues.extend(self._discover_queues(queue_manager, regex))
+        elif config.queue_patterns:
+            for pattern in config.queue_patterns:
+                queues.extend(self._discover_queues(queue_manager, pattern))
+
+        if config.queue_regex:
```

Comment:
This happens regardless of autodiscovery?

Reply:
The configurations `auto_discover_queues`, `queue_patterns`, `queue_regex` are not intended to be used together. Seems there is no point to use them together, either you want everything using `auto_discover_queues`, or match by pattern using `queue_patterns` or `queue_regex`

I though about raising an error, but want to avoid changing the existing behavior for `auto_discover_queues` and `queue_patterns`.

But I can maybe log a Warning if more than one of those 3 configs are used  `auto_discover_queues`, `queue_patterns`, `queue_regex`

Reply:
Yes, that is a good idea

Reply:
warning added